teuthology/task/install/rpm.py

```diff
@@ -18,49 +17,40 @@ def _remove(ctx, config, remote, rpm):
     :param remote: the teuthology.orchestra.remote.Remote object
     :param rpm: list of packages names to remove
     """
-    rpm = _package_overrides(rpm, remote.os)
+    remote_os = remote.os
+    dist_release = remote_os.name
+    rpm = _package_overrides(rpm, remote_os)
     log.info("Removing packages: {pkglist} on rpm system.".format(
         pkglist=", ".join(rpm)))
-    builder = _get_builder_project(ctx, remote, config)
-    dist_release = builder.dist_release
+    repos = config.get('repos')
 
     if dist_release in ['opensuse', 'sle']:
-        pkg_mng_cmd = 'zypper'
-        pkg_mng_opts = '-n --no-gpg-checks'
-        pkg_mng_subcommand_opts = '--capability'
+        remote.run(args='''
+            for d in {rpms} ; do
+                sudo zypper -n --no-gpg-checks remove --capability $d || true
+            done'''.format(rpms=' '.join(rpm)))
+        remote.run(args='sudo zypper clean -a')
     else:
-        pkg_mng_cmd = 'yum'
-        pkg_mng_opts = '-y'
-        pkg_mng_subcommand_opts = ''
+        remote.run(args='''
+            for d in {rpms} ; do
+                sudo yum -y remove $d || true
```

Comment:
same problem here - the run.Raw() calls have disappeared and this is a breaking change

Reply:
Look for comment for previous case.

Reply:
OK, that's interesting. Maybe it works when there is an existing SSH connection? I have personally experienced a failure and if you grep/ack the teuthology codebase for `run.Raw` there are many, many instances. Are you saying all these are unnecessary?

Reply:
The run.Raw absolutely necessary when you use `args` options as an array because each element has got quoted with _pipe_, but you don't need to quote special control combinations like. But if you use `args` parameter as a basestring it is not needed. I suppose authors of those many lines with run.Raw definitely prefer to pass bare arrays instead " ".join(options), but using all those run.Raw and long name variable makes code readability similar to assembler and way-way cryptic than many perl constructions. I think 'args' usage is originally designed to be a list of command arguments, not a shell script, and all those Raw injects are more like workaround. I don't know what was the reason, but probably they tried to reduce time and traffic packing script snippet in one exec_command invocation.